packages/ckeditor5-dev-env/lib/release-tools/utils/getnewreleasetype.js

```diff
@@ -5,7 +5,8 @@
 
 'use strict';
 
-const { commitTypes, transform } = require( '../changelog/writer-options' );
+const { availableCommitTypes } = require( './transform-commit-utils' );
+const transformCommitForCKEditor5Package = require( './transformcommitforckeditor5package' );
```

Comment:
So, if you have YAGNI now, then this entire refactoring may be YAGNI ;). It's the same category of problem - do we need to parametrize the `transformCommit`?

But, it'd also be more clear whether that specific function can be used directly here if the code was structured more clearly. As I mentioned before, you should physically separate it into general release helpers, tasks and helpers for CKEditor 5 packages and tasks and helpers for CKEditor 5 dev stuff. You'd then better see where this function landed. If in the first category, it should not know about `transformCommitForCKEditor5Package`.